src/modeler/AutoRest.Swagger/Validation/Core/RuleContext.cs

```diff
@@ -19,12 +19,16 @@ public class RuleContext
         /// Initializes a top level context for rules
         /// </summary>
         /// <param name="root"></param>
-        public RuleContext(ServiceDefinition root, Uri file) : this(null)
+        public RuleContext(ServiceDefinition root, Uri file, ServiceDefinitionMetadata metadata) : this(null)
         {
             this.Root = root;
             this.Value = root;
             this.File = file;
-            PopulateResourceTypes(root);
+            if (metadata.ServiceDefinitionDocumentType == ServiceDefinitionDocumentType.ARM)
```

Comment:
further observation: this `==` will fail if `metadata.ServiceDefinitionDocumentType` are multiple types at once!
Instead of duplicating the logic for document type filtering, I'd honestly just make `ResourceModels` a lazily initialized property! If someone asks for it, populate it, otherwise don't. That way, you automatically inherit the correct behavior from the fact that the rules are classified correctly.